### ISSUE TITLE
regex too strict

### ISSUE THREAD

**Issue by ClasherKasten** [Athena-Chaos-Driven-Code/AthenaDocumentor#19]
https://github.com/Athena-Chaos-Driven-Code/AthenaDocumentor/blob/98de48d73c4a5bcca6d43734c443fe78de039506/src/AthenaDocumentor/data/regex_expressions.py#L15

This expression is too strict, as tuple literals should be usable as default argument.

```python
def f(x = ()):
    ...
```

**Comment by AndreasSas** [comment]
Good call, didn't think of this, as I rarely use tuples as defaults for args/kwargs

**Comment by ClasherKasten** [comment]
> Good call, didn't think of this, as I rarely use tuples as defaults for args/kwargs

I didn't wrote this for `*args`, as `*args` usually don't have a default value, but more for parameters that have a type like `tuple`, `Sequence` and such and have as default value a tuple assigned to them.

**Comment by AndreasSas** [comment]
I didn't mean `*args` or `**kwargs` haha, I just meant arguments or keyword arguments in general

I think the following regex could fix it? 
`\(([^()]+.*)\)`
https://regexr.com/71ld7

**Comment by ClasherKasten** [comment]
> I think the following regex could fix it?
\(([^()]+.*)\)

Theoretically yes, but I think best option would be to switch to `ast.FunctionDef` as it provides all necessary information about parameters like types, default values, ...

**Comment by AndreasSas** [comment]
> > I think the following regex could fix it?
> > (([^()]+.*))
> 
> Theoretically yes, but I think best option would be to switch to `ast.FunctionDef` as it provides all necessary information about parameters like types, default values, ...

You are somewhat right yeah
I'm replacing the old regex with the `ast.unparse(node.args)` and `ast.unparse(node.returns)`
